### PR TITLE
[SIEM][CEF] Add support for Check Point devices

### LINKED ISSUE #16041: [Filebeat] Add module for Checkpoint Firewall
Most likely using Checkpoint Log Exporter and CEF format

Checkpoint can export logs via CEF using their "Log Exporter".  I found a list of mappings:
https://community.checkpoint.com/t5/Logging-and-Reporting/Log-Exporter-CEF-Field-Mappings/td-p/41060

Info on Log Exporter:
https://community.checkpoint.com/t5/Logging-and-Reporting/Log-Exporter-guide/td-p/9035

https://sc1.checkpoint.com/documents/R80.20.M2/WebAdminGuides/EN/CP_R80.20_M2_LoggingAndMonitoring_AdminGuide/html_frameset.htm?topic=documents/R80.20.M2/WebAdminGuides/EN/CP_R80.20_M2_LoggingAndMonitoring_AdminGuide/203039

### PR DESCRIPTION
**Pull request by adriansr**
This PR adds a new ingest pipeline and fields to populate from Check Point CEF logs.

Additional changes to the ECS processor:
- Make CEF key translation case-insensitive.
- Add missing CEF extension `c6a2`/`deviceCustomIPv6Address2`.
- Add deprecated `flexNumber[12]` extensions. Used by Check Point. Dropped from CEF after V23.
- Change case of `DeviceCustomNumber2` to align with the other fields.

Closes #16041

### CODE REVIEW

File: x-pack/filebeat/module/cef/log/_meta/fields.yml [hunk 1]
```diff
@@ -8,3 +8,253 @@
...
+    - name: sequencenum
+      type: keyword
+
+    - name: service_id
+      type: keyword
+
+    - name: version
+      type: keyword
+
+# TODO: Update to ECS 1.5 and remove.
```
Comment (adriansr) [inline x-pack/filebeat/module/cef/log/_meta/fields.yml]: Will remove and rebase once #16900 is merged

File: filebeat/docs/modules/cef.asciidoc [hunk 2]
```diff
@@ -52,6 +52,90 @@ configuring the SMC.  Testing was done with CEF logs from SMC version
...
+[float]
+==== Check Point devices
+
+This module will parse CEF data form Check Point devices as documented in
+https://community.checkpoint.com/t5/Logging-and-Reporting/Log-Exporter-CEF-Field-Mappings/td-p/41060[Log Exporter CEF Field Mappings.]
+
+Check Point CEF extensions are mapped as follows:
+[options="header"]
+|=======================================================================================================================
+| CEF Extension              | CEF Label value             | ECS Fields               | Non-ECS Field                  |
```
Comment (andrewkroh) [inline filebeat/docs/modules/cef.asciidoc]: This table is awesome. Helps me a lot in digesting the pipeline.

**Comment by adriansr** [comment]
@leehinman , I've attempted some categorisation by looking at the fields present. However, it is difficult to do anything acceptable without sample logs (I couldn't find any).

For example for event.action (or outcome) I only have the value "Allowed" in the sample logs but I don't know what other possible values are there so I'm reluctant to map everything else to `denied`. 

Have a look at the changes, I'm open to suggestions

**Comment by leehinman** [comment]
> @leehinman , I've attempted some categorisation by looking at the fields present. However, it is difficult to do anything acceptable without sample logs (I couldn't find any).
> 
> For example for event.action (or outcome) I only have the value "Allowed" in the sample logs but I don't know what other possible values are there so I'm reluctant to map everything else to `denied`.
> 
> Have a look at the changes, I'm open to suggestions

changes look great.  hopefully we can get more examples in the future.

**Pull request merged**